lib/middleware/route-option.js

```diff
@@ -0,0 +1,9 @@
+/**
+ * Routes the request based on action `type`, `callback_id`, and `name`
+ */
+module.exports = function routeOption(req, res, next) {
+  req.log.debug({ ...req.body, token: undefined }, 'Option load received from Slack');
+
+  req.url += `:${req.body.type}:${req.body.callback_id}:${req.body.name}`;
```

Comment:
I don't think we need to do this now, but I'm starting to wonder if we should extract some kind of "parameter router" middleware instead of mutating the `req.url` everywhere.

Something like this:

```js
router.post('/actions', withParams({type: 'message_action', callback_id: 'attach-to-issue'}, (req, res) => {
  // route handler here
})
```

It's a little more verbose, but would be more generic/reusable.

Reply:
Yeah, I like that a lot! Makes it much easier to understand what's going on especially for folks new to the codebase